File: examples/node/custom-model-training/dataset/asticaDataset_class_remove.js
const axios = require('axios');

const astica_endpoint = 'https://train.astica.ai/model/class_remove';
const astica_api_key = 'API KEY HERE'; //visit https://astica.ai
const astica_class_id = 0; //class_id to remove

const astica_requestData = {
    'tkn': astica_api_key,
    'class_id': astica_class_id,
};

axios.post(astica_endpoint, astica_requestData, { headers: { 'Content-Type': 'application/json' } })
.then((response) => {
    const data = response.data;
    // raw output
    console.log(data);
    // handle response
    if (data.status !== 'OK') {
        console.error('API request failed');
        console.error(data.error);
    } else {
        console.log('Class has been removed');
    }
})
.catch((error) => {
    console.error('Unable to remove class');
    console.error('Error:', error.toString());
});

File: examples/python/custom-model-training/dataset/asticaDataset_class_remove.py
import requests
import json

astica_endpoint = 'https://train.astica.ai/model/class_remove'
astica_api_key = 'API KEY HERE' #visit https://astica.ai
astica_class_id = 0; #class_id to remove

astica_requestData = {
    'tkn': astica_api_key,
    'class_id': astica_class_id,
}
try:
    response = requests.post(astica_endpoint, data=json.dumps(astica_requestData), headers={'Content-Type': 'application/json'})
    response.raise_for_status()  # Raise an exception if the request was unsuccessful
    data = json.loads(response.text)    
    #raw output
    print(data)
    #handle response
    if data['status'] != 'OK':
        print("api request failed")
        print(data["error"])
    else:
        print("Class has been removed")        
        
except requests.exceptions.RequestException as e:
    print("Unable to remove class")
    print("Error:", str(e))